mw_basic_auth.go

```diff
@@ -77,6 +77,11 @@ func (k *BasicAuthKeyIsValid) requestForBasicAuth(w http.ResponseWriter, msg str
 func (k *BasicAuthKeyIsValid) basicAuthHeaderCredentials(w http.ResponseWriter, r *http.Request) (username, password string, err error, code int) {
 	token := r.Header.Get("Authorization")
 	logger := k.Logger().WithField("key", obfuscateKey(token))
+	if token == "" && k.Spec.BasicAuth.ExtractFromBody {
```

Comment:
this edge case is a good candidate to cover by test

Reply:
Seems that we already have tests with empty headers + body extraction enabled in `TestBasicAuthFromBody`.

Reply:
why that test wasn't failing without this fix then?